app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js

```diff
@@ -297,6 +297,10 @@ window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
             }
           }
         }
+      },
+
+      applyRedactionIfRequired: function (PIIRemover, element, data) {
+        return element.hasAttribute('data-ga4-do-not-redact') ? data : PIIRemover.stripPIIWithOverride(data, true, true)
```

Comment:
Should we use `element.closest` instead? This would allow the element to check itself and it's parent elements for the attribute. This would account for clicking on a child link within content, and would allow you to put the attribute on the parent div instead of on every individual link on a page.

Reply:
Thanks @AshGDS, good idea - I've made that change and I've also added another test to check that this is working when `data-ga4-do-not-redact` is applied directly on an element and on a parent element.